script/ensure-go-installed

```diff
@@ -1,13 +1,13 @@
 #!/bin/bash
 
-PREFERRED_GO_VERSION=go1.17.11
-SUPPORTED_GO_VERSIONS='go1.1[567]'
+PREFERRED_GO_VERSION=go1.21.5
+SUPPORTED_GO_VERSIONS='go1.[1-2][01789]'
```

Comment:
This is horrible, but it will allow `go1.20`-`.21` and `go1.17`-`.19` to passthrough